packages/frontend/src/hooks/useSwapToast.tsx

```diff
@@ -0,0 +1,110 @@
+import { showToast, Toast, updateToast } from '@sifi/shared-ui';
+import { getEvmTxUrl, getViemErrorMessage } from 'src/utils';
+import { usePublicClient } from 'wagmi';
+import { useSpaceTravel } from 'src/providers/SpaceTravelProvider';
+import { useSwapFormValues } from './useSwapFormValues';
+import { useSifi } from 'src/providers/SDKProvider';
+import { useRefetchBalances } from './useRefetchBalances';
+
+const useSwapToast = () => {
+  const sifi = useSifi();
+  const { fromChain, toChain } = useSwapFormValues();
+  const { refetchAllBalances } = useRefetchBalances();
+  const publicClient = usePublicClient({ chainId: fromChain.id });
+  const { setThrottle } = useSpaceTravel();
+
+  const showErrorToast = (error: any) => {
+    if (error instanceof Error) {
+      showToast({ text: getViemErrorMessage(error), type: 'error' });
+    } else {
+      console.error(error);
+    }
+  };
+
+  const showSwapToast = async ({ hash }: { hash: `0x${string}` }) => {
+    const isJump = fromChain.id !== toChain.id;
+    let explorerLink: string | undefined;
+    if (isJump) {
+      explorerLink = `https://layerzeroscan.com/tx/${hash}`;
+    } else {
+      explorerLink = fromChain ? getEvmTxUrl(fromChain, hash) : undefined;
+    }
+
+    const swapToastId = showToast({
+      text: `${isJump ? 'Jump' : 'Swap'} initiated. Hold tight.`,
+      type: 'loading',
+      hideCloseIcon: true,
+      closeOnClick: false,
+      autoClose: false,
+    });
+
+    await publicClient.waitForTransactionReceipt({ hash });
+    refetchAllBalances();
+
+    // TODO: What if the transaction fails?
+
+    if (isJump) {
+      const intervalId = setInterval(async () => {
```

Comment:
This could probably be done in a cleaner manner with react-query, not essential

Reply:
Yes. Could we do the polling through React Query, and update the toast through a useEffect, or through a react-query callback if one is available?

Reply:
We can do this refactor later.